### PR TITLE
Added static method to build Permissions from permission Bitmasks

### PR DESCRIPTION
**Pull request by burdoto**
This change was proposed in the [Discord channel](https://discordapp.com/channels/151037561152733184/151326093482262528/551843038947966986).

### CODE REVIEW

**Comment by Saladoc** [comment]
I think this would better be suited as a `Permissions#fromBitmask(int)` method.

**Comment by burdoto** [comment]
Moved the method to `Permissions` 👍

**Comment by Saladoc** [comment]
You could have made it a static constructor method directly creating a `Permissions` object... Which was the whole point of me suggesting it.

**Comment by burdoto** [comment]
I'm not sure if that is better than returning a plain `Set`, as a `Permissions` object is more useful for a Collective of allowed, neutral and denied permissions.
However if that is requested, I can change it to be such a method.

**Comment by Saladoc** [comment]
When someone has a bitmask, where do they want to go from there? They will, in 99% of use cases, want to arrive at a `Permissions` object. So that's what should be provided. You can even extend the signature to include a denied bitmask as well, with 0 as a default parameter.

**Comment by burdoto** [comment]
That is a good idea. I will make those changes asap 👍

**Comment by burdoto** [comment]
Ready for new revision 👍

File: javacord-api/src/main/java/org/javacord/api/entity/permission/Permissions.java [hunk 1]
```diff
@@ -82,4 +82,43 @@ default PermissionsBuilder toBuilder() {
      */
     boolean isEmpty(); // We could check it in a default method, but it's faster to just check it in the implementation
 
+    /**
+     * Creates a {@code Permissions} object from the given bitmask.
+     * Permissions that are not included are marked as {@link PermissionState#UNSET}.
+     *
+     * @param bitmask The bitmask of allowed permissions.
+     * @return A {@code Permissions} objects created after the given bitmask.
```
Comment (Bastian) [inline javacord-api/src/main/java/org/javacord/api/entity/permission/Permissions.java]:
```suggestion
     * @return A {@code Permissions} object created after the given bitmask.
```
Just a small typo

File: javacord-api/src/main/java/org/javacord/api/entity/permission/Permissions.java [hunk 2]
```diff
@@ -82,4 +82,43 @@ default PermissionsBuilder toBuilder() {
...
+
+    /**
+     * Creates a {@code Permissions} object from the given bitmasks.
+     * Permissions that are not included in any bitmap are marked as {@link PermissionState#UNSET}.
+     * If a permission is included in both the allowed and denied bitmap,
+     * it is marked as {@link PermissionState#UNSET}.
+     *
+     * @param allowedBitmask The bitmask of allowed permissions.
+     * @param deniedBitmask The bitmask of denied permissions.
+     * @return A {@code Permissions} objects created after the given bitmasks.
```
Comment (Bastian) [inline javacord-api/src/main/java/org/javacord/api/entity/permission/Permissions.java]:
```suggestion
     * @return A {@code Permissions} object created after the given bitmasks.
```

File: javacord-api/src/main/java/org/javacord/api/entity/permission/Permissions.java [hunk 3]
```diff
@@ -82,4 +82,43 @@ default PermissionsBuilder toBuilder() {
...
+     *
+     * @param bitmask The bitmask of allowed permissions.
+     * @return A {@code Permissions} objects created after the given bitmask.
+     */
+    static Permissions fromBitmask(int bitmask) {
+        return fromBitmask(bitmask, 0);
+    }
+
+    /**
+     * Creates a {@code Permissions} object from the given bitmasks.
```
Comment (Bastian) [inline javacord-api/src/main/java/org/javacord/api/entity/permission/Permissions.java]:
```suggestion
     * Creates a {@code Permissions} object from the given bitmasks.
     * <p>
```

**Comment by burdoto** [comment]
All changes have been applied 👍

**Comment by Bastian** [comment]
Thx

**Pull request merged**